### DISCUSSION TITLE
Problem with emojis (prompt) squeezed

### DISCUSSION THREAD

**Discussion by invisible1988** [fish-shell/fish-shell#9158] [Q&A] [answered]
Hello!

I was playing a bit with fish and zsh. Now I've decided to go with fish. But I don't know why but I get Problems with the emoji width in my prompt.

I've already tested to set fish_emoji_width to 2. Before I had installed zsh + ohmyzsh. Now I'm not sure if there are some system configs or something left that create this issue. Has someone maybe an idea?

I would appericate it.

Bye bye

**Comment by faho** [comment, marked as answer]
This is about the *rendering*. That's up to the terminal, fish has no influence on that.

>I've already tested to set fish_emoji_width to 2

Setting $fish_emoji_width to 2 tells fish that the terminal will render emoji with a width of 2 columns (so it can figure out what column the prompt ends in). It doesn't *give* them a width of 2. That's up to the terminal.

Plus your issue here isn't that these are 1 column instead of 2, it's the position *inside* the column.

----

You have four solutions here:

1. Check your terminal settings to see if something can fix this - maybe you need a different font
2. Report a bug to your terminal
3. Switch terminals
4. Pick different characters

Fish can't do anything about this.
